src/lib/FlightTasks/tasks/Orbit/FlightTaskOrbit_params.c

```diff
@@ -0,0 +1,87 @@
+/****************************************************************************
+ *
+ *   Copyright (c) 2019 PX4 Development Team. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in
+ *    the documentation and/or other materials provided with the
+ *    distribution.
+ * 3. Neither the name PX4 nor the names of its contributors may be
+ *    used to endorse or promote products derived from this software
+ *    without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ * FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ * COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+ * OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ ****************************************************************************/
+
+/**
+ * @file FlightTaksOrbit_params.c
+ * Orbit specific parameters.
+ *
+ * @author Dennis Mannhart <dennis.mannhart@gmail.com>
+ */
+
+/**
+ * Centripetal maximum acceleration
+ *
+ * @unit m/s/s
+ * @min 1.0
+ * @max 10.0
+ * @increment 0.5
+ * @decimal 1
+ * @group FlightTaskOrbit
+ */
+PARAM_DEFINE_FLOAT(FLT_ORB_A_MAX, 2.0f);
```

Comment:
Couldn't we use `MPC_ACC_HOR(_MAX)?` here?

Reply:
There are two issues using the same parameter
1. `FLT_ORB_A_MAX` is the centripetal acceleration and `MPC_ACC_HOR(_MAX)` is more meant for linear acceleration (which basically defines the velocity setpoint ramp)
2. they have different default values. `MPC_ACC_HOR_MAX` is set to 5, and up to now the centripetal acceleration used in orbit was set to 2.

Although I also support in having fewer parameters, at the same time I see a problem with having one parameter for several different modes. For instance, `MPC_ACC_HOR` is now used for `FlightTaskAutoLine`, `FlightTaskAutoLineSmoothVel`, `FlightTaskManualPosition` and `FlightTaskManualPositionSmoothVel`. Since this parameter is still a tuning parameter defining the setpoint ramp, it might very well be the case that different flight-modes require a different value of this parameter.

Reply:
I don't see a difference between "centripetal acceleration" and "linear acceleration", it's the same thing. Also, since that value is used to saturate the speed and radius, using the `MPC_ACC_HOR` (used in all auto modes, default value of 3m/s2) seems safe. 

I only see the point if for some reason you want to have less acceleration during orbit than during normal flight, but why would you like that?

Reply:
> I don't see a difference between "centripetal acceleration" and "linear acceleration", it's the same thing

not really. linear acceleration MPC_ACC_HOR (at least that's was the case when I added it) was to control the velocity-ramp along the line during Auto. There is no ramp-limit perpendicular to the line. For Manual Position-control, it is different, but then there again I consider it as a tuning parameter to achieve a certain user-feel.

 "centripetal acceleration", on the other hand, is more about the change in velocity-vector.

From a testing perspective, all I know is that we currently use MPC_ACC_HOR = 6 and the orbit centripetal-acceleration was set to 2. If we combine both into one parameter, then this requires additional testing.

Reply:
> I don't see a difference between "centripetal acceleration" and "linear acceleration",

It's physically the same but in reality there's a big difference just because of the timespan the drone usually applies them. In an Orbit if you pull the radius smaller you will hit the acceleration limit faster than most people might expect and if the radius is the only thing you change you will stay accelerating with this maximum value 100% of the orbit time. 2m/s is my experimentally found value that also makes sense to fly with a bigger drone. A racer could easily just burn its battery with "double the thrust" but with a bigger drone you might want quick brakes from high speeds that have short acceleration periods but not extended periods of time spent with high thrust values.